src/views/AddItem.jsx

```diff
@@ -1,3 +1,111 @@
-export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+import './AddItem.css';
+import React, { useState } from 'react';
+import { addItem } from '../api';
+
+export function AddItem({ listToken }) {
+	const [itemData, setItemData] = useState({
+		itemName: '',
+		daysUntilNextPurchase: '7',
+	});
+	const [success, setSuccess] = useState('');
+	const [errorMsg, setErrorMsg] = useState('');
+	const [isAdded, setIsAdded] = useState(false);
+
+	function handleChange(event) {
+		const { name, value } = event.target;
+		setItemData((prevFormData) => {
+			return {
+				...prevFormData,
+				[name]: value,
+			};
+		});
+	}
+
+	async function handleSubmit(e) {
+		e.preventDefault();
+		try {
+			await addItem(listToken, itemData);
+			setIsAdded(true);
+			setSuccess('Data added successfully');
+		} catch (error) {
+			setIsAdded(false);
+			setErrorMsg('Adding data failed');
+		}
+
+		setItemData({
+			...itemData,
```

Comment:
This destructuring here is not needed as you rewrite them in the two lines bellow

Reply:
Updated